README.md

```diff
@@ -0,0 +1,99 @@
+# Background:
```

Comment:
We shouldn't add this content to main README. Considering this is a proposal, we should probably create a new dir (say `designs`) to put all the proposal documents.
The main README would consist of documentation, usage, contributing guide, etc
I would also suggest to split the proposal document and implementation into separate PRs so that they can be reviewed and merged separately

Reply:
Done. Design proposal is in this PR https://github.com/kanisterio/errkit/pull/2

Current PR's readme file has been updated to contain documentation only.